src/domains/ValuesDomain.js

```diff
@@ -42,6 +69,185 @@ export default class ValuesDomain {
     return this._elements;
   }
 
+  // return a set of values that may be result of performing the given operation on each pair in the
+  // Cartesian product of the value sets of the operands.
+  static binaryOp(realm: Realm, op: BabelBinaryOperator, left: ValuesDomain, right: ValuesDomain): ValuesDomain {
+    let leftElements = left._elements;
+    let rightElements = right._elements;
+    if (!leftElements || !rightElements || leftElements.size > 100 || rightElements.size > 100)
```

Comment:
I thought you prefer `=== undefined` over `!`?

Reply:
What's the reasoning behind the `> 100`? I have no clue, and don't see a comment.

Reply:
The ugliness of the pretty version of line 77 made me swallow ! for this instance.

I'll add a comment. The reason for the size restriction is that the larger a value set becomes, the less useful it becomes while at the same time it becomes computationally expensive.